sdk/src/main/java/io/opentdf/platform/sdk/Config.java

```diff
@@ -79,4 +83,63 @@ public static Consumer<TDFConfig> withMetaData(String metaData) {
     public static Consumer<TDFConfig> withSegmentSize(int size) {
         return (TDFConfig config) -> config.defaultSegmentSize = size;
     }
+
+    public static class NanoTDFConfig {
+        public ECCMode eccMode;
+        public NanoTDFType.Cipher cipher;
+        public SymmetricAndPayloadConfig config;
+        public List<String> attributes;
+        public List<KASInfo> kasInfoList;
+
+        public NanoTDFConfig() {
+            this.eccMode = new ECCMode();
+            this.eccMode.setEllipticCurve(NanoTDFType.ECCurve.SECP256R1);
+            this.eccMode.setECDSABinding(false);
+
+            this.cipher = NanoTDFType.Cipher.AES_256_GCM_96_TAG;
+
+            this.config = new SymmetricAndPayloadConfig();
+            this.config.setHasSignature(false);
+            this.config.setSymmetricCipherType(NanoTDFType.Cipher.AES_256_GCM_96_TAG);
+
+            this.attributes = new ArrayList<>();
+            this.kasInfoList = new ArrayList<>();
+        }
+    }
+
+    public static NanoTDFConfig newNanoTDFConfig(Consumer<NanoTDFConfig>... options) {
+        NanoTDFConfig config = new NanoTDFConfig();
+        for (Consumer<NanoTDFConfig> option : options) {
+            option.accept(config);
+        }
+        return config;
+    }
+
+    public static Consumer<NanoTDFConfig> witDataAttributes(String... attributes) {
+        return (NanoTDFConfig config) -> {
+            Collections.addAll(config.attributes, attributes);
+        };
+    }
+
+    public static Consumer<NanoTDFConfig> withNanoKasInformation(KASInfo... kasInfoList) {
+        return (NanoTDFConfig config) -> {
+            Collections.addAll(config.kasInfoList, kasInfoList);
+        };
+    }
+
+    public static Consumer<NanoTDFConfig> withEllipticCurve(String curve) {
+        NanoTDFType.ECCurve ecCurve;
+        if (curve.compareToIgnoreCase(NanoTDFType.ECCurve.SECP384R1.toString()) == 0) {
+            ecCurve = NanoTDFType.ECCurve.SECP384R1;
+        } else if (curve.compareToIgnoreCase(NanoTDFType.ECCurve.SECP521R1.toString()) == 0) {
+            ecCurve = NanoTDFType.ECCurve.SECP521R1;
+        } else {
```

Comment:
should have at least a warning here that the supplied curve string was not recognized before using the default.

Reply:
I agree. If no curve is provided, we should default, but if an invalid curve was supplied we should fail.